### DISCUSSION TITLE
Regarding Option() for Slash Command Paremeters

### DISCUSSION THREAD

**Discussion by Rwarcards762** [Pycord-Development/pycord#687] [Q&A] [answered]
Hello! Recently working on migrating over from `discord.py` after seeing that it is no longer being maintained, and working on moving my bot to the slash-based command interactions.

I have found the new `Option()` method for defining arguments in commands particularly interesting, and Discord seems to mention the ability to require different types of objects (such as Users, Channels, etc.) This would make life SO much easier for argument parsing.

However, I cannot for the life of me find any specific documentation about what actual types of data are supported for those `Option()` calls and was asking to see if there is a definitive list of options (other than just basic types like `str`, `int`, but more Discord-based requirements like `USER` and `CHANNEL`) and how one would use them?

Thanks in advance! I look forward to seeing how Pycord develops!

**Comment by Lulalaby** [comment, marked as answer]
The documentation is WIP rn for application command stuff. If you want to view the wip docs: https://docs.pycord.dev/en/document-app-commands/

**Reply by Rwarcards762** [reply]
Thanks for the quick reply. I've been trying to peruse through there to find something specific regarding Options() for command arguments (via `from discord.commands import Option`) and can't find anything -- I assume it just isn't in the docs yet. Is there any sort of information source containing info on this yet? Don't wanna start developing until I can take full advantage of easy argument parsing like that.

**Reply by Rwarcards762** [reply]
I seem to have found [this page](https://docs.pycord.dev/en/document-app-commands/api.html#discord.SlashCommandOptionType) but cannot seem to find how those are actually used as part of a call to `Option()`, ie `(argname: Option(TYPE_NAME, "Description")` -- how that corresponds to that `TYPE_NAME`.

**Reply by krittick** [reply]
argname: Option(discord.User, "Choose a user")

argname: Option(discord.TextChannel, "choose a channel")

Is that the sort of clarification you're looking for?

This section in enums might also help: https://github.com/Pycord-Development/pycord/blob/640c73ad5a5e664ace7d563e1d9575f50ea3707c/discord/enums.py#L622

**Reply by Rwarcards762** [reply]
Ah yes, okay. So it's just calls to `discord.TYPE` or simple types such as `int`, `str`. Makes sense.

That is the clarification I was looking for, and am just wondering if there was a definitive list of "options" that are allowed -- or if that docs page I mentioned above is the "definitive list."

Thank you!

**Reply by krittick** [reply]
The enums list I linked is the "definitive" one for the library. Otherwise you can find discord's official list here: https://discord.com/developers/docs/interactions/application-commands#application-command-object-application-command-option-type
